ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -316,23 +314,36 @@ protected void handleIncomingPacket(final Endpoint sourceEndpoint, final Packet
    * @return host address as string
    */
   static String deriveHost(final Endpoint sourceEndpoint, final Packet packet) {
-    return packet
-        .getPacketData(PingPacketData.class)
-        .flatMap(PingPacketData::getFrom)
-        .map(Endpoint::getHost)
-        .filter(
-            fromAddr ->
-                (!PING_PACKET_SOURCE_IGNORED.contains(fromAddr)
-                    && InetAddresses.isInetAddress(fromAddr)))
+    final Optional<String> pingPacketHost =
+        packet
+            .getPacketData(PingPacketData.class)
+            .flatMap(PingPacketData::getFrom)
+            .map(Endpoint::getHost);
+
+    return pingPacketHost
+        // fall back to source endpoint "from" if ping packet from address does not satisfy filters
+        .filter(InetAddresses::isInetAddress)
+        .filter(h -> !NetworkUtility.isUnspecifiedAddress(h))
+        .filter(h -> !NetworkUtility.isLocalhostAddress(h))
+        .filter(h -> isIpv6Available || !NetworkUtility.isIpV6Address(h))
         .stream()
         .peek(
             h ->
-                LOG.trace(
-                    "Using \"From\" endpoint {} specified in ping packet. Ignoring UDP source host {}",
-                    h,
-                    sourceEndpoint.getHost()))
+                LOG.atTrace()
+                    .setMessage(
+                        "Using \"From\" endpoint {} specified in ping packet. Ignoring UDP source host {}")
+                    .addArgument(h)
+                    .addArgument(sourceEndpoint::getHost)
+                    .log())
         .findFirst()
-        .orElseGet(sourceEndpoint::getHost);
+        .orElseGet(
+            () -> {
+              LOG.trace(
```

Comment:
LOG.atTrace would be nice ...

Reply:
since this is already in a lambda, the extra log lambda isn't necessary :)

Reply:
oh wait, yeah.  you are right!  I originally had this logging every time, but it was noisy and moved it to trace